lib/pdk/cli/util/changelog_generator.rb

```diff
@@ -0,0 +1,58 @@
+module PDK
+  module CLI
+    module Util
+      class ChangelogGenerator
+        def generate_changelog
+          retries = 0
+          result = nil
+          loop do
+            result = system('bundle exec rake changelog')
+            retries += 1
+            break if result || retries > 3
+          end
+
+          unless result
+            raise 'Error generating changelog'
+          end
+
+          output = File.read('CHANGELOG.md')
+
+          raise 'Uncategorized PRs; Please go label them' if output =~ %r{UNCATEGORIZED PRS; GO LABEL THEM}
+        end
+
+        def get_next_version(current_version)
+          current_version_map = current_version.split('.').map { |v| v.to_i }
+          puts '# Extracting target version from CHANGELOG.md'
```

Comment:
Should be using the PDK logging classes, not `puts`

Reply:
Will change this to PDK logger